source/module_base/test/complexarray_test.cpp

```diff
@@ -84,14 +107,78 @@ class ComplexArray_test : public testing::Test
 
 };
 
-TEST_F(ComplexArray_test,operator_equal)
+
+TEST(ComplexArray, construction1_test)
+{
+    ModuleBase::ComplexArray a(1,2,3,4);
+    ASSERT_EQ(a.getSize(),24);
+    ASSERT_EQ(a.getBound1(),1);
+    ASSERT_EQ(a.getBound2(),2);
+    ASSERT_EQ(a.getBound3(),3);
+    ASSERT_EQ(a.getBound4(),4);
+    /*
+    ASSERT_DEATH(ModuleBase::ComplexArray a(0,2,3,4),"");
+    ASSERT_DEATH(ModuleBase::ComplexArray a(1,0,3,4),"");
+    ASSERT_DEATH(ModuleBase::ComplexArray a(1,2,0,4),"");
+    ASSERT_DEATH(ModuleBase::ComplexArray a(1,2,3,0),"");
+    */
+}
+
+TEST(ComplexArray, construction2_test)
+{
+    ModuleBase::ComplexArray a(1,2,3,4);
+    ModuleBase::ComplexArray b(a);
+    ASSERT_EQ(b.getSize(),24);
+    ASSERT_EQ(b.getBound1(),1);
+    ASSERT_EQ(b.getBound2(),2);
+    ASSERT_EQ(b.getBound3(),3);
+    ASSERT_EQ(b.getBound4(),4);
+       
+    ASSERT_EQ(a.getSize(),24);
+    ASSERT_EQ(a.getBound1(),1);
+    ASSERT_EQ(a.getBound2(),2);
+    ASSERT_EQ(a.getBound3(),3);
+    ASSERT_EQ(a.getBound4(),4);
+
+}
+
+TEST(ComplexArray, construction3_test)
+{
+    ModuleBase::ComplexArray b(ModuleBase::ComplexArray(1,2,3,4));
+    ASSERT_EQ(b.getSize(),24);
+    ASSERT_EQ(b.getBound1(),1);
+    ASSERT_EQ(b.getBound2(),2);
+    ASSERT_EQ(b.getBound3(),3);
+    ASSERT_EQ(b.getBound4(),4);    
+}
+
+
+TEST_F(ComplexArray_test,operator_equal_complex)
 {   
-    c2 = a2; //c2 is just constructed as a CompleArray, but not assigned by any value
-    b2 = a2; //b2 is constructed as a ComplexArray, and is also assigned by a complex
-    EXPECT_EQ(c2,a2);
-    EXPECT_EQ(b2,a2);
-    EXPECT_NE(c2,a4);
-    EXPECT_DEATH(c2=a4,"");
+    for(int i = 0;i<a2.getSize();++i)
```

Comment:
Maybe add some comments here to notify that `a2=com1` was performed when constructing test class. Or just do it again.